### PR TITLE
Add Queryset-level delete

### LINKED ISSUE #10: Queryset-level deletes

### PR DESCRIPTION
**Pull request by Lynskylate**
Fixes #10 
implement Queryset-level delete
I want to reuse build_select_expression, but it seems to no way to change the expression's select column.
The return value of delete method is worth discussing.
Please review

### CODE REVIEW

**Comment by rafalp** [comment]
Does this change make it possible to explicitly delete all rows in the table? Django provides `.all().delete()` for this case and while it's rare, it is useful in some situations.

**Comment by Lynskylate** [comment]
> Does this change make it possible to explicitly delete all rows in the table? Django provides `.all().delete()` for this case and while it's rare, it is useful in some situations.

Sorry, we can't.  We need to refactor the QuerySet's method all to implement it.

File: tests/test_models.py [hunk 1]
```diff
@@ -190,3 +190,40 @@ def test_model_pk():
...
+        await User.objects.create(name="Tom")
+        await User.objects.create(name="Jany")
+        await User.objects.create(name="Lucy")
+
+        await User.objects.filter(name="Tom").delete()
+        assert await User.objects.count() == 2
+        await User.objects.delete(name="Jany")
+        assert await User.objects.count() == 1
+
+        await Product.objects.create(name="T-Shirt", rating=5)
```
Comment (dopry) [inline tests/test_models.py]: Can we add `await Product.objects.create(name="Dress", rating=4)` to ensure both filter clauses are respected by `build_delete_expression`

**Comment by aminalaee** [comment]
This is way too old, and I think #111 is simpler, so I'll close this for now.

**Pull request closed without merging**